features/machine/machine_healthcheck.feature

```diff
@@ -130,3 +130,40 @@ Feature: MachineHealthCheck Test Scenarios
       | mhc-<%= machine_set.name %>-1: total targets: 1,  maxUnhealthy: 0, unhealthy: 1. Short-circuiting remediation   |
       | mhc-<%= machine_set.name %>-2: total targets: 1,  maxUnhealthy: 90%, unhealthy: 1. Short-circuiting remediation |
     """
+
+  # @author miyadav@redhat.com
+  # @case_id OCP-28718
+  @admin
+  @destructive
+  Scenario: [MHC] - Machine Node startup timeout should be configurable
+    Given I have an IPI deployment
+    And I switch to cluster admin pseudo user
+    Then I use the "openshift-machine-api" project
+    And admin ensures machine number is restored after scenario
+
+    Given I clone a machineset and name it "machineset-clone-28718"
+
+    # Create MHC with configurable node startup timeout
+    When I run oc create over "<%= BushSlicer::HOME %>/testdata/cloud/mhc/mhc_configurabletimeout.yaml" replacing paths:
+      | n                                                                                  | openshift-machine-api       |
+      | ["metadata"]["name"]                                                               | mhc-<%= machine_set.name %> |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-cluster"]    | <%= machine_set.cluster %>  |
+      | ["spec"]["selector"]["matchLabels"]["machine.openshift.io/cluster-api-machineset"] | <%= machine_set.name %>     |
+      | ["spec"]["nodeStartupTimeout" ]                                                    | 15m                         |
+
+    Given I create the 'Ready' unhealthyCondition
+    Then a pod becomes ready with labels:
+     | api=clusterapi, k8s-app=controller |
+
+    And I wait up to 600 seconds for the steps to pass:
+    """
+     When I run the :logs admin command with:
+     | resource_name | <%= pod.name %>                |
+     | c             | machine-healthcheck-controller |
+    Then the output should contain:
+     | Ensuring a requeue happens in 15m0  |
+    """
+   And I ensure "mhc-<%= machine_set.name %>" machinehealthcheck is deleted after scenario
```

Comment:
this line should be in line 153, in case the above step failed, this step will never be executed

Reply:
Thank you , made the change  , looks good..